graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java

```diff
@@ -6298,20 +6314,34 @@ private void addTemplateLiteralString(final ArrayList<Expression> rawStrings, fi
      *      ModuleItemList
      * </pre>
      */
-    private FunctionNode module(final String moduleName) {
+    private FunctionNode module(final String moduleName, boolean moduleBlock) {
         // Make a pseudo-token for the script holding its start and length.
         int functionStart = Math.min(Token.descPosition(Token.withDelimiter(token)), finish);
         final long functionToken = Token.toDesc(FUNCTION, functionStart, source.getLength() - functionStart);
         final int functionLine = line;
 
         final Scope moduleScope = Scope.createModule();
-        final IdentNode ident = null;
-        final ParserContextFunctionNode script = createParserContextFunctionNode(
-                        ident,
-                        functionToken,
-                        FunctionNode.IS_MODULE,
-                        functionLine,
-                        Collections.<IdentNode> emptyList(), 0, moduleScope);
+        final ParserContextFunctionNode script;
+
+        final IdentNode ident = moduleBlock ? new IdentNode(functionToken, finish, moduleName) : null;
+
+        if (moduleBlock) {
+            script = createParserContextFunctionNode(
+                            ident,
+                            functionToken,
+                            FunctionNode.IS_MODULE_BLOCK,
```

Comment:
In that case it would suffice to avoid the `if` but have a `moduleBlock?FunctionNode.IS_MODULE_BLOCK:FunctionNode.IS_MODULE` here, right? no other difference?

Reply:
Yes, that is correct, adjusted the code to your suggestion.